frontend/model/state.js

```diff
@@ -28,12 +29,12 @@ var store // this is set and made the default export at the bottom of the file.
 // 'state' is the Vuex state object, and it can only store JSON-like data
 var dropAllMessagesUntilRefresh = false
 
-const CONTRACT_REGEX = /^gi\.contracts\/(?:([^/]+)\/)(?:([^/]+)\/)?process$/
 // guard all sbp calls for contract actions with this function
-export function selectorIsContractOrAction (sel: string) {
-  if (!CONTRACT_REGEX.test(sel)) {
-    throw new GIErrorIgnoreAndBanIfGroup(`bad selector '${sel}' for contract type!`)
+function guardedSBP (sel: string, ...data: any): any {
```

Comment:
shouldn't we name this `guardedSelector`?  I'm not sure if I'll be clear but... SBP is the paradigm we use only, not "the thing" we are guarding...

Reply:
It's called `guardedSBP` because it's exactly the same as `sbp` except it guards against non-whitelisted selectors. So it's a guarded call to `sbp`, hence `guardedSBP`.